source/secure_tunneling.c

```diff
@@ -2547,6 +2547,16 @@ int aws_secure_tunnel_send_message(
         message_op->options_storage.storage_view.connection_id = 0;
     }
 
+    if (!s_aws_secure_tunnel_protocol_version_match_check(secure_tunnel, &message_op->options_storage.storage_view)) {
```

Comment:
In the event of a missmatch `s_aws_secure_tunnel_protocol_version_match_check` will log "id=%p: Secure Tunnel is currently using Protocol V%d and received a message using Protocol V%d" before returning false. This will be confusing in the logs as this check is now occurring on an outbound message. We could add a bool to the function to determine if the message being checked is outbound/inbound and then adjust the log accordingly there. It's also good to log in that function since we'll get more detail in the log (which version is established and which version is attempted to be used)

Reply:
Maybe it'll be better if we use a log message without a message origin in `s_aws_secure_tunnel_protocol_version_match_check`? Something like `WARN("Protocol Version mismatch: Secure Tunnel is currently using Protocol V%d and a message is using Protocol V%d")`.
Code will be less verbose while all the info will still be in logs because we log additionally which kind of message caused this warning.